crates/vsock/src/thread_backend.rs

```diff
@@ -307,11 +307,7 @@ impl VsockThreadBackend {
         self.backend_rxq
             .push_back(ConnMapKey::new(pkt.dst_port(), pkt.src_port()));
 
-        self.stream_map.insert(
-            stream_fd,
-            // SAFETY: Safe as the file descriptor is guaranteed to be valid.
```

Comment:
It does not immediately become clear to me what constraint was violated here. The docs of `from_raw_fd` also only state that "it should be valid", not that it must not be tied to another UnixStream instance? Or is that not a safety constraint, but a functional requirement (as in: "it won't crash, but also won't do what you want")?

Maybe some detail about this would be helpful in the commit message.

Reply:
Yeah, I just wonder why the safety requirement does not spell that out, but I assume it is because it does not cause any undefined behaviour but just causes undesired errors?

Reply:
Because that depends on the type implementing the trait, and in this case `UnixStream` assumes it owns the file descriptor.